server/src/main/java/org/opensearch/common/cache/service/NodeCacheStats.java

```diff
@@ -77,4 +80,10 @@ public boolean equals(Object o) {
     public int hashCode() {
         return Objects.hash(statsByCache, flags);
     }
+
+    // Get the immutable cache stats for a given cache, used to avoid having to process XContent in tests.
+    // Safe to expose publicly as the ImmutableCacheStatsHolder can't be modified after its creation.
+    public ImmutableCacheStatsHolder getStatsByCache(CacheType cacheType) {
+        return statsByCache.get(cacheType);
+    }
```

Comment:
Is this method required for verification within the tests? Probably not the right thing to do, although not sure if there is better way of achieving this

Reply:
It's not technically required, but I think it's worth adding because it simplifies the tests a lot. 

Before I added this, I accomplished the same thing by calling NodeCacheStats.toXContent(), converting that to a nested Map<Object, String>, and traversing through that recursively. It worked but was pretty gross. But I noticed no other tests seem to directly handle XContent like this, and instead they use methods to access whatever data is used to produce XContent in the first place. So I thought we could simplify the tests by adding this.